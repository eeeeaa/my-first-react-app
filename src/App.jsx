import { NavBar } from "./topSection/navigationBar.jsx";
import { useState } from "react";
import { Bio } from "./content/bio.jsx";
import "./App.css";

function App() {
  const [currentPage, setCurrentPage] = useState(<Bio />);
  const [backgroundColor, setBackgroundColor] = useState("#242424");

  const onNavButtonClick = (currentPage) => {
    setCurrentPage(currentPage);
  };

  const onColorChangeClick = (backgroundColor) => {
    setBackgroundColor(backgroundColor);
  };
  return (
    <div className="appLayout" style={{ backgroundColor }}>
      <NavBar
        handleNavClick={onNavButtonClick}
        handleColorChangeClick={onColorChangeClick}
      />
      <div className="content">{currentPage}</div>
    </div>
  );
}

export default App;
