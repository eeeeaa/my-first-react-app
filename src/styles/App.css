.appLayout {
  height: 100vh;
}

.content {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

@keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

.fade-in {
  animation: fadeIn 0.5s ease;
}
