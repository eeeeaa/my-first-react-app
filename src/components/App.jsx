import { NavBar } from "./navigationBar.jsx";
import { useState } from "react";
import { Bio } from "./bio.jsx";
import "../styles/App.css";

function App() {
  const [currentPage, setCurrentPage] = useState(<Bio />);
  const [backgroundColor, setBackgroundColor] = useState("#242424");

  const onNavButtonClick = (currentPage) => {
    setCurrentPage(currentPage);
  };

  const onColorChangeClick = (backgroundColor) => {
    setBackgroundColor(backgroundColor);
  };
  return (
    <div
      className="appLayout"
      style={{ backgroundColor }}
      data-testid="main-app-page"
    >
      <NavBar
        handleNavClick={onNavButtonClick}
        handleColorChangeClick={onColorChangeClick}
      />
      <div className="content">{currentPage}</div>
    </div>
  );
}

export default App;
