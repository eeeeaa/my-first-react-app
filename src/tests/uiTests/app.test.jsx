import { render, screen } from "@testing-library/react";
import App from "../../components/App.jsx";
import { describe, it, expect } from "vitest";
import userEvent from "@testing-library/user-event";

describe("App component", () => {
  it("renders App", () => {
    // since screen does not have the container property, we'll destructure render to obtain a container for this test
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
  it("change background color and button text on random color button click", async () => {
    const user = userEvent.setup();

    render(<App />);
    const button = screen.getByRole("button", { name: "Random Color (0)" });

    await user.click(button);

    expect(button.textContent).toMatch("Random Color (1)");
  });
});
