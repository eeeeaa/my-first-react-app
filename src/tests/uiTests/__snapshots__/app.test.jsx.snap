// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders App 1`] = `
<div>
  <div
    class="appLayout"
    data-testid="main-app-page"
    style="background-color: rgb(36, 36, 36);"
  >
    <div
      class="nav-bar"
    >
      <div
        class="nav-logo"
      >
        Dummy Corp
      </div>
      <div
        class="nav-buttons"
      >
        <button>
          Random Color (
          0
          )
        </button>
        <button>
          Bio
        </button>
        <button>
          Example
        </button>
        <button>
          Person
        </button>
        <button>
          Mock Chatroom
        </button>
        <button>
          Legacy Example
        </button>
      </div>
    </div>
    <div
      class="content"
    >
      <div
        class="bio-container"
      >
        <div
          class="intro"
        >
          <h1>
            Welcome to my website!
          </h1>
        </div>
        <p
          class="summary"
        >
          You can find 
          Leeroy
          's thoughts here.
        </p>
        <div
          class="card"
        >
          <img
            alt="Leeroy"
            class="avatar"
            height="64"
            src="https://www.shutterstock.com/image-vector/illustration-retro-style-city-pixel-600nw-2289504657.jpg"
            width="64"
          />
        </div>
        <div>
          <h2>
            Here's a list of things about me
          </h2>
          <ul
            class="bio-list"
          >
            <li>
              I love animals
            </li>
            <li>
              I hate veggies
            </li>
          </ul>
        </div>
        <div>
          <h2>
            Here's a list of things of my favorite animals
          </h2>
          <ul
            class="bio-list"
          >
            <li>
              dog
            </li>
            <li>
              cat
            </li>
            <li>
              bird
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App component > renders magnificent monkeys 1`] = `
<div>
  <div
    class="appLayout"
    data-testid="main-app-page"
    style="background-color: rgb(36, 36, 36);"
  >
    <div
      class="nav-bar"
    >
      <div
        class="nav-logo"
      >
        Dummy Corp
      </div>
      <div
        class="nav-buttons"
      >
        <button>
          Random Color (
          0
          )
        </button>
        <button>
          Bio
        </button>
        <button>
          Example
        </button>
        <button>
          Person
        </button>
        <button>
          Mock Chatroom
        </button>
        <button>
          Legacy Example
        </button>
      </div>
    </div>
    <div
      class="content"
    >
      <div
        class="bio-container"
      >
        <div
          class="intro"
        >
          <h1>
            Welcome to my website!
          </h1>
        </div>
        <p
          class="summary"
        >
          You can find 
          Leeroy
          's thoughts here.
        </p>
        <div
          class="card"
        >
          <img
            alt="Leeroy"
            class="avatar"
            height="64"
            src="https://www.shutterstock.com/image-vector/illustration-retro-style-city-pixel-600nw-2289504657.jpg"
            width="64"
          />
        </div>
        <div>
          <h2>
            Here's a list of things about me
          </h2>
          <ul
            class="bio-list"
          >
            <li>
              I love animals
            </li>
            <li>
              I hate veggies
            </li>
          </ul>
        </div>
        <div>
          <h2>
            Here's a list of things of my favorite animals
          </h2>
          <ul
            class="bio-list"
          >
            <li>
              dog
            </li>
            <li>
              cat
            </li>
            <li>
              bird
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
